src/app_service/service/node_configuration/service.nim

```diff
@@ -73,45 +98,65 @@ method setWakuVersion*(self: Service, wakuVersion: int) =
-  var dataDir = currentNetworkDetails.config.dataDir
+  var dataDir = currentNetworkDetails.config.DataDir
   dataDir.removeSuffix("_rpc")
 
   var newConfiguration = self.configuration
-  newConfiguration.NetworkId = currentNetworkDetails.config.networkId
+  newConfiguration.NetworkId = currentNetworkDetails.config.NetworkId
   newConfiguration.DataDir = dataDir
-  newConfiguration.UpstreamConfig.Enabled = currentNetworkDetails.config.upstreamConfig.enabled
-  newConfiguration.UpstreamConfig.URL = currentNetworkDetails.config.upstreamConfig.url
-  self.saveConfiguration(newConfiguration)
+  newConfiguration.UpstreamConfig.Enabled = currentNetworkDetails.config.UpstreamConfig.Enabled
+  newConfiguration.UpstreamConfig.URL = currentNetworkDetails.config.UpstreamConfig.URL
+  return self.saveConfiguration(newConfiguration)
   
-method setBloomFilterMode*(self: Service, bloomFilterMode: bool) =
+method setBloomFilterMode*(self: Service, bloomFilterMode: bool): bool =
   if(not self.settingsService.saveWakuBloomFilterMode(bloomFilterMode)):
     error "error saving waku bloom filter mode ", methodName="setBloomFilterMode"
-    return
+    return false
 
   var newConfiguration = self.configuration
   newConfiguration.WakuConfig.BloomFilterMode = bloomFilterMode
-  self.saveConfiguration(newConfiguration)
+  return self.saveConfiguration(newConfiguration)
+
+method setBloomLevel*(self: Service, bloomLevel: string): bool =
+  # default is BLOOM_LEVEL_NORMAL
+  var bloomFilterMode = false
+  var fullNode = true
+  if (bloomLevel == BLOOM_LEVEL_LIGHT):
+    bloomFilterMode = false
+    fullNode = false
+  elif (bloomLevel == BLOOM_LEVEL_FULL):
+    bloomFilterMode = true
+    fullNode = true
 
-method setBloomLevel*(self: Service, bloomFilterMode: bool, fullNode: bool) =
   if(not self.settingsService.saveWakuBloomFilterMode(bloomFilterMode)):
     error "error saving waku bloom filter mode ", methodName="setBloomLevel"
     return
 
   var newConfiguration = self.configuration
-  newConfiguration.WakuConfig.BloomFilterMode = bloomFilterMode
-  newConfiguration.WakuConfig.FullNode = fullNode
-  newConfiguration.WakuConfig.LightClient = not fullNode
-  self.saveConfiguration(newConfiguration)
 
-method setFleet*(self: Service, fleet: string) =
+  ## In the old code we didn't take into consideration waku version when we stored bloom level,
+  ## now we have that part added, this need to be checked.
+  let wakuVersion = self.getWakuVersion()
+  if wakuVersion == WAKU_VERSION_1:
+    newConfiguration.WakuConfig.BloomFilterMode = bloomFilterMode
+    newConfiguration.WakuConfig.FullNode = fullNode
+    newConfiguration.WakuConfig.LightClient = not fullNode
+  elif wakuVersion == WAKU_VERSION_2:
+    newConfiguration.WakuV2Config.BloomFilterMode = bloomFilterMode
+    newConfiguration.WakuV2Config.FullNode = fullNode
+    newConfiguration.WakuV2Config.LightClient = not fullNode
+  
+  return self.saveConfiguration(newConfiguration)
+
```

Comment:
BloomLevel only applies to wakuV1, the condition for WAKU_VERSION_2 is not necessary.
Perhaps an exception must be thrown if the function is executed when waku2 is enabled?